edgelet/hsm-sys/azure-iot-hsm-c/tests/test_utils/test_utils.c

```diff
@@ -0,0 +1,193 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#include <stdlib.h>
+
+#include "testrunnerswitcher.h"
+#include "azure_c_shared_utility/crt_abstractions.h"
+#include "azure_c_shared_utility/strings.h"
+#include "azure_c_shared_utility/uniqueid.h"
+
+#if (defined __WINDOWS__ || defined _WIN32 || defined _WIN64 || defined _Windows)
+    #include <direct.h>
+    #include <intsafe.h>
+    #include <windows.h>
+
+    #define SLASH "\\"
+#else
+    #include <sys/stat.h>
+    #include <sys/types.h>
+    #include <unistd.h>
+
+    #define SLASH "/"
+#endif
+
+#define UID_SIZE            37
+#define MAX_FILE_NAME_SIZE  256
+#define CREATE_DIR_OK       0
+#define CREATE_DIR_EXISTS   1
+#define CREATE_DIR_ERROR    2
+#define MAX_ATTEMPTS        10
+
+static char* get_temp_base_dir(void)
+{
+    char *result = calloc(MAX_FILE_NAME_SIZE, 1);
+    ASSERT_IS_NOT_NULL(result);
+
+#if (defined __WINDOWS__ || defined _WIN32 || defined _WIN64 || defined _Windows)
+    DWORD count = GetTempPathA(MAX_FILE_NAME_SIZE, result);
+    ASSERT_IS_TRUE_WITH_MSG(count < MAX_FILE_NAME_SIZE, "TestUtil Line:" TOSTRING(__LINE__));
+#else
+    strcpy_s(result, MAX_FILE_NAME_SIZE, "/tmp/");
+#endif
+    ASSERT_ARE_NOT_EQUAL_WITH_MSG(size_t, 0, strlen(result), "TestUtil Line:" TOSTRING(__LINE__));
+
+    return result;
+}
+
+static int make_test_dir(const char* dir_path)
+{
+    int status, result;
+#if defined __WINDOWS__ || defined _WIN32 || defined _WIN64 || defined _Windows
+    status = _mkdir(dir_path);
+#else
+    status = mkdir(dir_path, S_IRWXU | S_IRGRP | S_IXGRP | S_IROTH | S_IXOTH);
+#endif
+
+    if (status != 0)
+    {
+        if (errno == EEXIST)
+        {
+            printf("Directory '%s' already exists.\r\n", dir_path);
+            result = CREATE_DIR_EXISTS;
+        }
+        else
+        {
+            printf("Directory create failed for '%s'. Errno: %s.\r\n", dir_path, strerror(errno));
+            result = CREATE_DIR_ERROR;
+        }
+    }
+    else
+    {
+        result = CREATE_DIR_OK;
+    }
+
+    return result;
+}
+
+char *create_temp_dir_path(const char *dir_guid)
+{
+    int status;
+    char *tmp_dir, *dir_path;
+
+    tmp_dir = get_temp_base_dir();
+    dir_path = calloc(MAX_FILE_NAME_SIZE, 1);
+    ASSERT_IS_NOT_NULL_WITH_MSG(dir_path, "TestUtil Line:" TOSTRING(__LINE__));
+    status = snprintf(dir_path, MAX_FILE_NAME_SIZE, "%shsm_test_%s", tmp_dir, dir_guid);
+    ASSERT_IS_TRUE_WITH_MSG(((status > 0) || (status < MAX_FILE_NAME_SIZE)), "TestUtil Line:" TOSTRING(__LINE__));
+    free(tmp_dir);
+
+    return dir_path;
+}
+
+char* hsm_test_util_create_temp_dir(char **dir_guid)
```

Comment:
>char **dir_guid [](start = 36, length = 15)

What's the reason for returning the guid? I didn't see you using it for anything during the test.

Reply:
Its used for deleting the test dir that was created using the GUID. I did not want to pass a dir path directly into the delete utility API for safety reasons.